src/core/inlines.js

```diff
@@ -129,20 +129,16 @@ export function run(conf) {
     if (subtxt.length === 1) continue;
 
     const df = document.createDocumentFragment();
-    while (subtxt.length) {
-      const t = subtxt.shift();
+    for (let i = 0; i < subtxt.length; i++) {
+      const t = subtxt[i];
       let matched = null;
-      if (subtxt.length) matched = subtxt.shift();
+      if (i != subtxt.length - 1) {
+        matched = subtxt[i + 1];
+        i++;
+      }
```

Comment:
I've heard `shift()` is fast enough in modern engines, and personally found it's more readable here. Thoughts? @marcoscaceres

Reply:
@saschanaz Yes, I was somewhat doubtful about the performance improvements (basically it would only matter if `subtxt` had some ~1000 elements, not really occuring in an actual spec). I will revert back to `shift()`.

Reply:
It be nice if we didn't need to `shift()` at all. Haven't looked at the code in detail in a while, so don't know if it's possible... but with thinking about.